src/main/java/seedu/address/model/InfoHandler.java

```diff
@@ -209,13 +212,33 @@ public VisitBook generateVisitsByPerson(Id personId) {
         ReadOnlyVisitBook tempVisitBook = model.getVisitBook();
         VisitBook visitsByPerson = new VisitBook();
         for (int i = 0; i < tempVisitBook.getVisitList().size(); i++) {
-            if (tempVisitBook.getVisitList().get(i).getPerson().getId().equals(personId)) {
+            Visit visit = tempVisitBook.getVisitList().get(i);
+            if (isValidVisit(visit, personId)) {
                 visitsByPerson.addVisit(tempVisitBook.getVisitList().get(i));
             }
         }
         return visitsByPerson;
     }
 
+    /**
+     * Checks if a visit is valid. It is considered valid if it meets 2 criteria.
+     * 1. It is by a person with a specified id.
+     * 2. It must also be within 2 weeks of the current date.
+     * Note: 2 weeks refers to a 14 day period that is inclusive of the current date, but exclusive of the date 14
+     * days ago.
+     * @param visit Visit being checked.
+     * @param personId Id of the person.
+     * @return A boolean of whether a visit is by a person with a specified id.
+     */
+    private boolean isValidVisit(Visit visit, Id personId) {
```

Comment:
The name is quite general since visits as a class may have their own definition of validity too. While the docs does give an explanation, method name itself is still not intuitive.

Reply:
Renamed!